### PR TITLE
refactor apple_min_version_flag()

### LINKED ISSUE #15494: [technical debt] Review the flow after using the `get_apple_sdk_fullname` helper
Using this reads weird:
- The ``get_apple_sdk_fullname`` returns a value with the sdk+sdk_version.
- But then, later, this value is passed to ``apple_min_version_flag()`` which requires the sdk_version separately, ignores the value in the ``os_sdk`` and use the ``os.version`` instead.

The whole flow is confusing (it is not because of this PR, it was already confusing)

_Originally posted by @memsharded in https://github.com/conan-io/conan/pull/15465#discussion_r1461171949_

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Omit
Docs: Omit

Close https://github.com/conan-io/conan/issues/15494

### CODE REVIEW

File: conan/tools/apple/apple.py [hunk 1]
```diff
@@ -57,8 +57,17 @@ def get_apple_sdk_fullname(conanfile):
...
-def apple_min_version_flag(os_version, os_sdk, subsystem):
+def apple_min_version_flag(conanfile):
     """compiler flag name which controls deployment target"""
+    os_ = conanfile.settings.get_safe('os')
+    os_sdk = conanfile.settings.get_safe('os.sdk')
+    os_sdk = os_sdk or ("macosx" if os_ == "Macos" else None)
+    if os_sdk is None:
+        raise ConanException("Please, specify a suitable value for os.sdk")
+
+    # FIXME: Why using os.version and not the os.sdk_version???
```
Comment (Neverminda) [inline conan/tools/apple/apple.py]: os.version specifies the deployment target, which is controlled by this min version flag.
You can create applications with the same min os version with different SDKs.

File: conan/tools/apple/apple.py [hunk 2]
```diff
@@ -57,39 +61,52 @@ def get_apple_sdk_fullname(conanfile):
...
         # need early return as string did not fit into the "normal" schema
-        return f'-target arm64-apple-xros{os_version}'
-    elif 'xrsimulator' in os_sdk:
+        "xros": f"-target arm64-apple-xros{os_version}",
         # need early return as string did not fit into the "normal" schema
-        return f'-target arm64-apple-xros{os_version}-simulator'
+        "xrsimulator": f"-target arm64-apple-xros{os_version}-simulator",
+    }.get(os_sdk,
+          # especial case, despite Catalyst is macOS, it requires an iOS version argument
+          '-mios-version-min' if subsystem == 'catalyst' else '')
```
Comment (memsharded) [inline conan/tools/apple/apple.py]: This is changing behavior, previously the ``catalyst`` was overriding the ``macos``, but no longer the case with this change

**Review by memsharded — COMMENTED** [review]
Looking good!

File: conans/test/unittests/util/apple_test.py [hunk 3]
```diff
@@ -82,40 +85,3 @@ def test_to_apple_arch(self):
         self.assertEqual(_to_apple_arch('armv8_32'), 'arm64_32')
         self.assertIsNone(_to_apple_arch('mips'))
         self.assertEqual(_to_apple_arch('mips', default='mips'), 'mips')
-
-    @pytest.mark.skipif(platform.system() != "Darwin", reason="Requires OSX")
-    def test_xcrun(self):
```
Comment (franramirez688) [inline conans/test/unittests/util/apple_test.py:87]: Moved to integration tests. It's running xcrun tool so, IMO, better to keep it there.

File: conans/test/unittests/util/apple_test.py [hunk 4]
```diff
@@ -48,11 +45,17 @@ class TestApple:
                                '-mtvos-simulator-version-min=10.1'),
                               ("Macos", "10.1", "macosx", "catalyst", '-mios-version-min=10.1'),
                               ("Solaris", "10.1", None, None, ''),
-                              ("Macos", "10.1", None, None, ''),
+                              ("Macos", "10.1", None, None, '-mmacosx-version-min=10.1'),
```
Comment (memsharded) [inline conans/test/unittests/util/apple_test.py:48]: So this was a bug before?

File: conans/test/unittests/client/toolchain/autotools/autotools_toolchain_test.py [hunk 5]
```diff
@@ -349,7 +349,7 @@ def test_crossbuild_from_macos_to_non_apple_os():
     conanfile.settings = MockSettings({"os": "Android", "arch": "armv8"})
     conanfile.settings_build = MockSettings({"os": "Macos", "arch": "armv8"})
     be = AutotoolsToolchain(conanfile)
-    assert be.apple_min_version_flag == ''
+    assert be.apple_min_version_flag is None
```
Comment (memsharded) [inline conans/test/unittests/client/toolchain/autotools/autotools_toolchain_test.py]: Are we sure this cannot break users?

File: conans/test/unittests/util/apple_test.py (hunk 4 above)
Comment (franramirez688) [reply conans/test/unittests/util/apple_test.py:48]: Yes, I think so.

File: conans/test/unittests/client/toolchain/autotools/autotools_toolchain_test.py (hunk 5 above)
Comment (franramirez688) [reply conans/test/unittests/client/toolchain/autotools/autotools_toolchain_test.py]: Let's prevent it.

File: conan/tools/apple/apple.py [hunk 6]
```diff
@@ -57,39 +61,51 @@ def get_apple_sdk_fullname(conanfile):
...
+    Gets the most common flags in Apple systems. If it's a cross-building context
+    SDK path is mandatory so if it could raise an exception if SDK is not found.
+
+    :param conanfile: <ConanFile> instance.
+    :param is_cross_building: boolean to indicate if it's a cross-building context.
+    :return: tuple of Apple flags (apple_min_version_flag, apple_arch, apple_isysroot_flag).
+    """
+    if not is_apple_os(conanfile):
+        # Keeping legacy defaults
+        return "", None, None
```
Comment (franramirez688) [inline conan/tools/apple/apple.py:99]: @memsharded keeping legacy behavior.

**Review by franramirez688 — APPROVED** [review]

**Review by memsharded — COMMENTED** [review]
LGTM

File: conan/tools/apple/apple.py (hunk 1 above)
Comment (memsharded) [reply conan/tools/apple/apple.py]: Thanks for the feedback!

**Pull request merged**